scRNA-seq/post-processing_scRNA-seq.Rmd

```diff
@@ -0,0 +1,150 @@
+---
+title: "scRNA-seq_post_processing"
+output: html_notebook
+---
+
+**CCDL 2019**
+
+In this notebook, we'll perform quality control analyses and normalization of 
+scRNA-seq count data. 
+
+As opposed to bulk rna-seq, there are there are a few main things to look out for
+in single cell rna-seq:
+
+**Single-cell rna-seq has...**  
+- More amplification (and more of it's associated biases and error)  
+- More 0's in the gene expression data (most genes aren't expressed across cell types)  
+
+This script is adapted from [Seurat pbmc tutorial](https://satijalab.org/seurat/pbmc3k_tutorial.html) 
+and from [COMBINE lab's alevin tutorial](https://combine-lab.github.io/alevin-tutorial/2018/alevin-seurat/)
+
+For these analyses, we will need Seurat and dplyr packages. 
+```{r Install Packages}
+# Attach library
+library(Seurat)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+#### Perform QC checks with 'csoneson/alevinQC'
+This will provide html output with graphs evaluating the data. Keep in mind that 
+the data we have processed in this workshop is not the full dataset, and won't 
+look as good as the full set.
+```{r AlevinQC }
+# Store the path to the alevin output files
+alevin.file <- file.path("data", "alevin_output", "pbmc_1k_v2_S1_L001_subset")
+
+# Produce a QC report
+alevinQC::alevinQCReport(alevin.file,
+                         sampleId = "pbmc_1k_v2_S1_L001_subset", 
+                         outputFile = "pbmc_1k_v2_S1_L001_qc_report.html", 
```

Comment:
Should this go somewhere like `results` or `data/qc` rather than the top directory?

Reply:
To my point about tracking this with git:
```
Error: The file .//pbmc_1k_v2_S1_L001_qc_report.html already exists. Please remove or rename the file, provide another value of outputFile, or set forceOverwrite = TRUE.
```

Reply:
Changed this to save to `data` as well as added the `forceOverwrite = TRUE` argument.

Reply:
I don't know if we necessarily want `forceOverwrite = TRUE`, we just shouldn't give participants this QC report. If we think about what we have here as a template or guide, setting `forceOverwrite = TRUE` is the kind of thing that could get folks into trouble (and probably exactly why the default behavior of this function is `forceOverwrite = FALSE`).

Reply:
Imagine I added more cells or something upstream and then forgot to change the output filenames. It would not get them into trouble _here_ but we should think more generally.

Reply:
Alrighty. Makes sense.